pykafka/producer.py

```diff
@@ -102,7 +124,32 @@ def __init__(self,
         self._retry_backoff_ms = retry_backoff_ms
         self._required_acks = required_acks
         self._ack_timeout_ms = ack_timeout_ms
-        self._batch_size = batch_size
+        self._max_queued_messages = max_queued_messages
+        self._min_queued_messages = min_queued_messages
+        self._linger_ms = linger_ms
+        self._block_on_queue_full = block_on_queue_full
+        self._worker_exception = None
+        self._worker_trace_logged = False
+        self._running = False
+        self.start()
+
+    def raise_worker_exceptions(fn):
```

Comment:
I've actually considered all of this, and have waffled back and forth between having it be a decorator and a simple two lines of code at the end of every function it's needed in. I liked the decorator approach better than calling a member function at the end of every function. I also considered only putting it on user-facing functions, but I chose to put it everywhere because of the possibility that users will call private functions. I couldn't find a case where having these checks on functions called on worker threads actually causes a problem - the thread will not have a chance to raise the exception a second time from the decorator, since it will already have exited after the original exception (from the `break` in its main loop error handler). I've also not been able to replicate a case where having `exit()` decorated causes a problem, but that's the one I'm least confident about.

Since a lot of what you've said here aligns with my gut instinct about this, I believe I'll keep the decorator on public methods only, unless I've inadvertently convinced you otherwise here.